lib/super_diff/rspec/monkey_patches.rb

```diff
@@ -736,6 +734,20 @@ def expected_action_for_failure_message
           def matcher_text_builder_class
             SuperDiff::RSpec::MatcherTextBuilders::RaiseError
           end
+
+          private
+
+          def describe_expected_error
+            if @expected_error.is_a? Class
+              @expected_error.name
+            elsif @expected_error.is_a? Regexp
+              "Exception #{description_of(@expected_error)}"
+            elsif @expected_error.respond_to? :description
+              @expected_error.description
+            else
+              SuperDiff.inspect_object(@expected_error, as_lines: false)
```

Comment:
Is this the right fallback?

Reply:
Hmm. We could do this, although I'm not confident `inspect_object` covers every case, so it could blow up if given something unexpected. Would `@expected_error.inspect` or `@expected_error.to_s` work?

Reply:
@mcmire I think so. In most cases, it would be the same. Three notable exceptions would be:

* there's an applicable inspection tree builder applying to the arg in `extra_inspection_tree_builders`
* the `CustomObject` inspector applies (i.e. arg responds to `#attributes_for_super_diff`)
* the `TimeLike` inspector applies, so output would be something like `2023-11-22 16:28:18 -0800` instead of `#<Time 2023-11-22 16:28:18 -08:00 (PST)>`, though it would be pretty odd if someone expected to raise a `Time` or `Date`

Thinking of it this way, maybe people would be surprised if SuperDiff basically ignored the `#attributes_for_super_diff` or even custom differ that they'd supplied. And in any case, if `SuperDiff.inspect_object` blows up on the argument, I'd wager that the current behavior – that is, calling `#name` on it – would almost certainly also blow up.

What do you think?

Reply:
I think it makes sense to inspect the object here. `inspect_object` is designed to fall back to a default implementation. RSpec uses a similar concept with `description_of` and ObjectFormatter, so it isn't too different to use our own formatter here.